src/masks/InteractionMasks.tsx

```diff
@@ -599,110 +349,84 @@ export default class InteractionMasks<R, K extends keyof R> extends React.Compon
       updated: { [cellKey]: value } as never,
       action: UpdateActions.COLUMN_FILL
     });
-  };
+  }
 
-  onCommit = ({ cellKey, rowIdx, updated }: CommitEvent<R>): void => {
-    this.props.onGridRowsUpdated({
+  function onCommit({ cellKey, rowIdx, updated }: CommitEvent<R>): void {
+    onGridRowsUpdated({
       cellKey,
       fromRow: rowIdx,
       toRow: rowIdx,
       updated,
       action: UpdateActions.CELL_UPDATE
     });
-    this.closeEditor();
-  };
-
-  onCommitCancel = (): void => {
-    this.closeEditor();
-  };
+    closeEditor();
+  }
 
-  getRowTop(rowIdx: number): number {
-    return rowIdx * this.props.rowHeight;
+  function onCommitCancel(): void {
+    closeEditor();
   }
 
-  getSelectedDimensions = (selectedPosition: Position): Dimension => {
-    const { columns, scrollLeft, rowHeight } = this.props;
-    const top = this.getRowTop(selectedPosition.rowIdx);
-    const dimension = getSelectedDimensions({ selectedPosition, columns, scrollLeft, rowHeight });
+  function getSelectedDimensions(selectedPosition: Position): Dimension {
+    const top = rowHeight * selectedPosition.rowIdx;
+    const dimension = getDimensions({ selectedPosition, columns, scrollLeft, rowHeight });
     dimension.top = top;
     return dimension;
-  };
+  }
 
-  renderSingleCellSelectView() {
+  function renderSingleCellSelectView() {
     return (
-      !this.state.isEditorEnabled && this.isGridSelected() && (
+      !isEditorEnabled && isCellWithinBounds(selectedPosition) && (
         <SelectionMask
-          {...this.getSelectedDimensions(this.state.selectedPosition)}
-          ref={this.selectionMask}
+          {...getSelectedDimensions(selectedPosition)}
+          ref={selectionMaskRef}
         >
-          {this.isDragEnabled() && (
+          {isDragEnabled() && (
             <DragHandle
-              onDragStart={this.handleDragStart}
-              onDragEnd={this.handleDragEnd}
-              onDoubleClick={this.onDragHandleDoubleClick}
+              onDragStart={handleDragStart}
+              onDragEnd={handleDragEnd}
+              onDoubleClick={onDragHandleDoubleClick}
             />
           )}
         </SelectionMask>
       )
     );
   }
 
-  renderCellRangeSelectView() {
-    const { columns, rowHeight } = this.props;
-    return (
-      <>
-        <SelectionRangeMask
-          {...getSelectedRangeDimensions({ selectedRange: this.state.selectedRange, columns, rowHeight })}
-        />
-        <SelectionMask
-          {...this.getSelectedDimensions(this.state.selectedPosition)}
-          ref={this.selectionMask}
+  const rowData = rowGetter(selectedPosition.rowIdx);
```

Comment:
Can we always expect `rowGetter` to handle `-1` index cases?

Reply:
We cannot. I have moved it so it is only called when the editor is enabled